python/ray/autoscaler/ray-schema.json

```diff
@@ -291,6 +291,33 @@
                        "resources": {
                            "type": "object",
                            ".*": {"type": "number"}
+                       },
+                       "initialization_commands": {
+                           "$ref": "#/definitions/commands",
+                           "description": "List of commands that will be run before `setup_commands`. If docker is enabled, these commands will run outside the container and before docker is setup."
+                       },
+                       "setup_commands": {
+                           "$ref": "#/definitions/commands",
+                           "description": "List of common shell commands to run to setup nodes."
+                       },
+                       "head_setup_commands": {
```

Comment:
Do we really need head_setup / head_start_ray commands? I think we can just remove these and only allow them to be specified in the main config. After all there is only 1 head node ever.

Reply:
personally, i think api consistency is worth it here

Reply:
Thinking about this more, I think the reason why to remove it is that users may be confused about where to put their configs. Can we actually do the following:
- Remove head_setup_* from the per-node commands
- Also remove worker_setup_*.

This would make it 100% consistent, since you can only specify head in the base commands, and there is neither head nor worker setup commands in the per-node commands. Btw, this is also how the original design had the config.